provider/azure/internal/azurecli/az_test.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package azurecli
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/juju/errors"
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+type azSuite struct {
+	testing.CleanupSuite
+}
+
+var _ = gc.Suite(&azSuite{})
+
+func (s *azSuite) TestExec(c *gc.C) {
+	executor := &testExecutor{
+		expectedCommand: []string{"az", "cmd", "-o", "json"},
+		result:          testValue{OK: true},
+	}
+	s.PatchValue(&newExecutor, executor.new)
+	var result testValue
+	err := Exec(&result, "cmd")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, jc.DeepEquals, testValue{true})
+}
+
+func (s *azSuite) TestExecError(c *gc.C) {
+	executor := &testExecutor{
+		expectedCommand: []string{"az", "cmd", "-o", "json"},
+		error:           errors.New("test error"),
+	}
+	s.PatchValue(&newExecutor, executor.new)
+	err := Exec(nil, "cmd")
+	c.Assert(err, gc.ErrorMatches, "execution failure: test error")
+}
+
+func (s *azSuite) TestExecJSONError(c *gc.C) {
+	executor := &testExecutor{
+		expectedCommand: []string{"az", "cmd", "-o", "json"},
+		result:          []byte("}"),
+	}
+	s.PatchValue(&newExecutor, executor.new)
+	err := Exec(nil, "cmd")
+	c.Assert(err, gc.ErrorMatches, "cannot unmarshal output: invalid character '}' looking for beginning of value")
+}
+
+func (s *azSuite) TestGetAccessToken(c *gc.C) {
+	tok := AccessToken{
+		AccessToken:  "`1234567890",
+		ExpiresOn:    "2017-06-05 10:20:43.752534",
+		Subscription: "00000000-0000-0000-0000-00000001",
+		Tenant:       "00000000-0000-0000-0000-00000002",
+		TokenType:    "Bearer",
+	}
+	executor := &testExecutor{
+		expectedCommand: []string{"az", "account", "get-access-token", "-o", "json"},
+		result:          tok,
```

Comment:
This seems a little too self-fulfilling to me. I think I'd prefer it if these tests contained actual
copy & pasted output from executions of the real commands - then we can have
a little more confidence that the json struct annotations and types are correct.